### PR TITLE
Migrating all LiveData to Coroutines StateFlow

### PR DESCRIPTION
**Pull request by guuilp**
## Implementation

I've changed all `MutableLiveData` to `MutableStateFlow`. There's one breaking change in this: `MutableStateFlow` does not accept null as the initial value. Because of that, I've created a new state for each ViewModel: `Initial`.

I thought about `Loading`, but as you don't have any loading, I chose `Initial`.

### CODE REVIEW

**Comment by guuilp** [comment]
Fixes #103

File: app/src/main/java/it/ilker/repeatcard/ui/quiz/QuizViewModel.kt [hunk 1]
```diff
@@ -1,16 +1,17 @@
...
 import it.ilker.repeatcard.db.FlashcardDatabase
 import it.ilker.repeatcard.db.flashcard.FlashcardRepository
 import it.ilker.repeatcard.models.question.Question
 import it.ilker.repeatcard.models.quizresult.QuizResult
 import it.ilker.repeatcard.ui.util.exhaustive
-import java.util.UUID
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.launch
+import java.util.*
```
Comment (Dement0) [inline app/src/main/java/it/ilker/repeatcard/ui/quiz/QuizViewModel.kt:14]: We should maintain the old import instead of * import.

**Review by Dement0 — APPROVED** [review]
Hey @guuilp awesome job! Just 2 things, if you can remove the * import and add your name (if you would like to) to contributors so that I don't have to make another PR just for these :rofl: 

BTW, I'd like to know what you think about the app (architecture, missing features, UI sucks etc...). If you 're interested in maybe we can implement those features together :guitar:

**Comment by Dement0** [comment]
> ## Implementation
> 
> I've changed all `MutableLiveData` to `MutableStateFlow`. There's one breaking change in this: `MutableStateFlow` does not accept null as the initial value. Because of that, I've created a new state for each ViewModel: `Initial`.
> 
> I thought about `Loading`, but as you don't have any loading, I chose `Initial`.

Yea `Loading` would be awesome as you have indicated. In fact, I wanted to implement that with a `ProgressBar` but I didn't have time.. Maybe I can create an issue for that :tada:

File: app/src/main/java/it/ilker/repeatcard/ui/quiz/QuizViewModel.kt (hunk 1 above)
Comment (guuilp) [reply app/src/main/java/it/ilker/repeatcard/ui/quiz/QuizViewModel.kt]: Fixed!

**Comment by guuilp** [comment]
> > ## Implementation
> > I've changed all `MutableLiveData` to `MutableStateFlow`. There's one breaking change in this: `MutableStateFlow` does not accept null as the initial value. Because of that, I've created a new state for each ViewModel: `Initial`.
> > I thought about `Loading`, but as you don't have any loading, I chose `Initial`.
> 
> Yea `Loading` would be awesome as you have indicated. In fact, I wanted to implement that with a `ProgressBar` but I didn't have time.. Maybe I can create an issue for that 🎉

Renamed all Initial state to Loading, so when implementing the Loading is easier 😄

**Comment by guuilp** [comment]
> Hey @guuilp awesome job! Just 2 things, if you can remove the * import and add your name (if you would like to) to contributors so that I don't have to make another PR just for these 🤣

Thanks! Fixed and added my user to the contributors 👍 

> BTW, I'd like to know what you think about the app (architecture, missing features, UI sucks etc...). If you 're interested in maybe we can implement those features together 🎸

About the architecture, I really liked it! Easy to understand and to maintain. I just missed unit tests, but that can also be a task.

About the features, I confess that I haven't heard about Flashcards. I read about them right after starting on your project. Because of that, I think I can't suggest new features now 😢 

About the UI, the internal screens (adding directory, adding flashcards) are not so beautiful, but I'm not a designer to suggest how to improve it 🤣 . I'd open some apps to give some insights about the design.

About helping with the project, I'm in!

**Review by Dement0 — APPROVED** [review]

**Pull request merged**